api/v1alpha1/onload_types.go

```diff
@@ -73,6 +73,19 @@ type OnloadSpec struct {
 	// ImagePullPolicy is the policy used when pulling images.
 	// More info: https://kubernetes.io/docs/concepts/containers/images#updating-images
 	ImagePullPolicy v1.PullPolicy `json:"imagePullPolicy,omitempty"`
+
+	// +optional
+	// ControlPlane allows fine-tuning of the Onload control plane server.
+	ControlPlane ControlPlaneSpec `json:"controlPlane"`
+}
+
+type ControlPlaneSpec struct {
+	// +optional
+	// Parameters to pass to the Onload control plane server when launched
+	// by the Onload kernel module. Technically, these values are appended
+	// to /sys/module/onload/parameters/cplane_server_params.
+	// +kubebuilder:default:={"-K"}
+	Parameters []string `json:"parameters"`
```

Comment:
Is it possible/desirable to pass an explicitly empty list here in case the user doesn't want `"-K"`?

Reply:
Yes, it's also covered in both unit test changes. 👍 

```diff
diff --git a/config/samples/onload/base/onload_v1alpha1_onload.yaml b/config/samples/onload/base/onload_v1alpha1_onload.yaml
index 4e5b981..6a8d495 100644
--- a/config/samples/onload/base/onload_v1alpha1_onload.yaml
+++ b/config/samples/onload/base/onload_v1alpha1_onload.yaml
@@ -50,6 +50,8 @@ spec:
     userImage: onload/onload-user
     version:
     imagePullPolicy: Always
+    controlPlane:
+      parameters: []
   devicePlugin:
     imagePullPolicy: Always
     # Limit number of pods that can request Onload on a node
```
Results in:
```console
$ oc describe pod onload-onload-device-plugin-ds-xl276 | grep -A8 'Environment:$'
    Environment:
      POD_NAME:                 onload-onload-device-plugin-ds-xl276 (v1:metadata.name)
      POD_NAMESPACE:            default (v1:metadata.namespace)
      CONTAINER_NAME:           onload-worker
      ONLOAD_CP_SERVER_PATH:    /mnt/onload/sbin/onload_cp_server
      ONLOAD_CP_SERVER_PARAMS:
    Mounts:
      /mnt/onload from worker-volume (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-jjkks (ro)
```

Eventually:
```console
$ cat /sys/module/onload/parameters/cplane_server_params
exec 6b4b87f2f483ead127f4adfe0229c4c8cc882a4b0c179772c466e1af6e636640 /mnt/onload/sbin/onload_cp_server
```